src/firmware/app/control/trajectory_planner_impl.h

```diff
@@ -0,0 +1,275 @@
+ *
+ * @param num_elements [in] The number of nodes(elements) of the generated trajectory
+ *
+ * @param x_values [out] The x value of each trajectory node. This array is of length
+ * num_elements.
+ *
+ * @param y_values [out] The y value of each trajectory node. THis array is of length
+ * num_elements
+ *
+ * @param segment_lengths [out] The length of each trajectory segment (euclidean of the XY
+ * 2d polynomial). This array is of length num_elements-1
+ */
+void app_trajectory_planner_generate2dSegmentNodesAndLengths_impl(
+    float t_start, float t_end, Polynomial2dOrder3_t path_2d, unsigned int num_elements,
+    float *x_values, float *y_values, float *segment_lengths);
+
+/**
+ * Function that modifies an existing speed profile to be backwards continuous. This means
+ * that it is possible to reach every speed node within the limits of deceleration
+ *
+ * @pre All arrays are pre-allocated up to at least TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ *
+ * @param segment_lengths [in] The length of each segment in the trajectory.  Units of
+ * parameters must be in consistent magnitudes, ie: distances in mm and speed in mm/s.
+ *
+ * @param max_allowable_acceleration [in] The maximum acceleration that can occur anywhere
+ * on the trajectory
+ *
+ * @param initial_speed [in] The initial speed at the beginning of the trajectory.  Units
+ * of parameters must be in consistent magnitudes, ie: distances in mm and acceleration in
+ * mm/s.
+ *
+ * @param num_segments [in] The number of nodes(elements) in the trajectory
+ *
+ * @param speeds [in/out] The existing speed profile to be modified in-place to be
+ * backwards continuous. This array is of length num_elements. The units of output are
+ * consistent with the input parameters.
+ *
+ * @return A status indicating whether or not modification was successful
+ */
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_modifySpeedsToBeBackwardsContinuous_impl(
+    unsigned int num_segments, float *segment_lengths, float max_allowable_acceleration,
+    float initial_speed, float *speeds);
+
+/**
+ * Function creates a forwards continuous speed profile based on the specified parameters.
+ * The profile will never exceed the maximums speeds specified by the
+ * max_allowable_speed_profile
+ *
+ * @pre All arrays must be pre-allocated up to at least
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ *
+ * @param segment_lengths [in] The length of each segment between data points on the speed
+ * profile. In meters.
+ *
+ * @param max_allowable_speed_profile [in] An array that limits the max speed at each
+ * point on the profile. Units of parameters must be in consistent magnitudes, ie:
+ * distances in mm and speed in mm/s.
+ *
+ * @param max_allowable_acceleration [in] The maximum allowable acceleration along any
+ * segment in the profile. Units of parameters must be in consistent magnitudes, ie:
+ * distances in mm and acceleration in mm/s^2.
```

Comment:
should be `ex`, not `ie`, they imply significantly different things here. Similarly for other params here

Reply:
Changed all occurrences of ie